src/Microsoft.ML.Data/DataLoadSave/Binary/Codecs.cs

```diff
@@ -773,7 +767,7 @@ public Reader(DateTimeZoneCodec codec, Stream stream, int items)
                     for (int i = 0; i < _entries; i++)
                     {
                         _ticks[i] = Reader.ReadInt64();
-                        Contracts.CheckDecode(_ticks[i] == DvInt8.RawNA || (ulong)_ticks[i] <= DvDateTime.MaxTicks);
+                        Contracts.CheckDecode((ulong)_ticks[i] <= DvDateTime.MaxTicks);
```

Comment:
>(ulong)_ticks[i] <= DvDateTime.MaxTicks [](start = 46, length = 39)

If you are writing the ticks, and if it were being stored as `long.MinValue`, then casting to `ulong` would definitely fail this test. So this is wrong here.

Reply:
Other serialization schemes and checks for the other similar types have similar failure modes.

---
In reply to: [212043046](https://github.com/dotnet/machinelearning/pull/683#discussion_r212043046) [](ancestors = 212043046)

Reply:
DvDateTime PR has the right serialization code but I will change this here. I didn't care too much here since the that PR will replace this code.

---
In reply to: [212043134](https://github.com/dotnet/machinelearning/pull/683#discussion_r212043134) [](ancestors = 212043134,212043046)